src/ray/core_worker/lib/golang/go_worker.cpp

```diff
@@ -0,0 +1,282 @@
+  int result_length = node_to_connect.length();
+  RAY_LOG(WARNING) << "got nodeinfo len:" << result_length;
+  *result = (char *)malloc(result_length + 1);
+  memcpy(*result, node_to_connect.c_str(), result_length);
+  return result_length;
+}
+
+__attribute__((visibility("default"))) int go_worker_CreateActor(char *type_name,
+                                                                 char **result) {
+  std::vector<std::string> function_descriptor_list = {type_name};
+  ray::FunctionDescriptor function_descriptor =
+      ray::FunctionDescriptorBuilder::FromVector(ray::rpc::GOLANG,
+                                                 function_descriptor_list);
+  ActorID actor_id;
+  ray::RayFunction ray_function = ray::RayFunction(ray::rpc::GOLANG, function_descriptor);
+  // TODO
+  std::string full_name = "";
+  std::string ray_namespace = "";
+  ray::ActorCreationOptions actor_creation_options{
+      0,
+      0,  // TODO: Allow setting max_task_retries from Java.
+      static_cast<int>(1),
+      {},
+      {},
+      {},
+      /*is_detached=*/false,
+      full_name,
+      ray_namespace,
+      /*is_asyncio=*/false};
+  auto status = ray::CoreWorkerProcess::GetCoreWorker().CreateActor(
+      ray_function, {}, actor_creation_options,
+      /*extension_data*/ "", &actor_id);
+  if (!status.ok()) {
+    RAY_LOG(FATAL) << "Failed to create actor:" << status.message()
+                   << " for:" << type_name;
+    return 0;
+  }
+  int result_length = actor_id.Size();
+  *result = (char *)malloc(result_length + 1);
+  memcpy(*result, (char *)actor_id.Data(), result_length);
+  return result_length;
+}
+
+__attribute__((visibility("default"))) int go_worker_SubmitActorTask(void *actor_id,
+                                                                     char *method_name,
+                                                                     int num_returns,
+                                                                     void **object_ids) {
+  auto actor_id_obj = ByteArrayToId<ray::ActorID>((char *)actor_id);
+  std::vector<std::string> function_descriptor_list = {method_name};
+  ray::FunctionDescriptor function_descriptor =
+      ray::FunctionDescriptorBuilder::FromVector(ray::rpc::GOLANG,
+                                                 function_descriptor_list);
+  ray::RayFunction ray_function = ray::RayFunction(ray::rpc::GOLANG, function_descriptor);
+  std::string name = "";
+  std::unordered_map<std::string, double> resources;
+  ray::TaskOptions task_options{name, num_returns, resources};
+
+  std::vector<ObjectID> obj_ids;
+  ray::CoreWorkerProcess::GetCoreWorker().SubmitActorTask(actor_id_obj, ray_function, {},
+                                                          task_options, &obj_ids);
+
+  int object_id_size = ObjectID::Size();
+  for (size_t i = 0; i < obj_ids.size(); i++) {
+    void *result = malloc(object_id_size);
+    memcpy(result, (char *)obj_ids[i].Data(), object_id_size);
```

Comment:
Should we use strings instead of bytes for the object id?

Reply:
I tried using strings, but it results in double freeing since the string is destructed by C++...